scio-avro/src/main/scala/com/spotify/scio/avro/syntax/SCollectionSyntax.scala

```diff
@@ -203,7 +205,73 @@ final class ProtobufSCollectionOps[T <: Message](private val self: SCollection[T
       shardNameTemplate,
       tempDirectory
     )
-    self.write(ProtobufIO[T](path))(param)
+    self.write(ProtobufObjectFileIO[T](path))(param)
+  }
+}
+
+final class TypedMagnolifyProtobufSCollectionOps[T](private val self: SCollection[T])
+    extends AnyVal {
+
+  /**
+   * Save this SCollection as a Protobuf file.
+   *
+   * Protobuf messages are serialized into `Array[Byte]` and stored in Avro files to leverage Avro's
+   * block file format.
+   */
+  def saveAsProtobufFile[U <: Message](
+    path: String,
+    numShards: Int = ProtobufTypedObjectFileIO.WriteParam.DefaultNumShards,
+    suffix: String = ProtobufTypedObjectFileIO.WriteParam.DefaultSuffixProtobuf,
+    codec: CodecFactory = ProtobufTypedObjectFileIO.WriteParam.DefaultCodec,
+    metadata: Map[String, AnyRef] = ProtobufTypedObjectFileIO.WriteParam.DefaultMetadata,
+    shardNameTemplate: String = ProtobufTypedObjectFileIO.WriteParam.DefaultShardNameTemplate,
+    tempDirectory: String = ProtobufTypedObjectFileIO.WriteParam.DefaultTempDirectory,
+    filenamePolicySupplier: FilenamePolicySupplier =
+      ProtobufTypedObjectFileIO.WriteParam.DefaultFilenamePolicySupplier,
+    prefix: String = ProtobufTypedObjectFileIO.WriteParam.DefaultPrefix
+  )(implicit ct: ClassTag[U], pt: ProtobufType[T, U]): ClosedTap[T] = {
+    implicit val tCoder: Coder[T] = self.coder
+    val param = ProtobufTypedObjectFileIO.WriteParam[GenericRecord](
+      numShards,
+      suffix,
+      codec,
+      metadata,
+      filenamePolicySupplier,
+      prefix,
+      shardNameTemplate,
+      tempDirectory
+    )
+    self.write(ProtobufTypedObjectFileIO[T, U](path))(param)
+  }
+}
+
+final class TypedMagnolifyAvroSCollectionOps[T](private val self: SCollection[T]) {
```

Comment:
can we make `T: AvroMagnolifyType` part of `TypedMagnolifyAvroSCollectionOps`'s context bound so that all methods have access to it?

Reply:
makes sense! I'm in favor of making APIs more IDE-friendly 👍